smf/switch_zone_setup/switch_zone_setup

```diff
@@ -1,19 +1,32 @@
-#!/bin/bash
+#!/bin/ksh
+
 set -ex -o pipefail
 
 . /lib/svc/share/smf_include.sh
 
-# set up the users required for wicket.
-USERS=(wicket support)
+# set up the users required for wicket and support.
+USERS=(
+    (user=wicket group=wicket gecos='Wicket User')
+    (user=support group=support gecos='Oxide Support'
+        profiles=('Primary Administrator')
+    )
+)
+
+for i in "${!USERS[@]}"; do
+    nameref u=USERS[$i]
 
-for user in ${USERS[@]}; do
-    if ! id -u $user >/dev/null 2>&1; then
-        # Add a new group for the user.
-        groupadd "$user"
-        # Add the user.
-        useradd -m -g "$user" "$user"
-        # Remove the password from the user to enable passwordless login.
-        passwd -d "$user"
+    # Add a new group for the user.
+    getent group "${u.group}" >/dev/null 2>&1 || groupadd "${u.group}"
+    # Add the user.
+    getent passwd "${u.user}" >/dev/null 2>&1 \
+        || useradd -m -g "${u.group}" -c "${u.gecos}" "${u.user}"
+    # Remove the password from the user to enable passwordless login.
+    passwd -d "${u.user}"
```

Comment:
I haven't opened a PR for this yet, but: we're going to want to keep `passwd -d` for wicket and change support to  `passwd -N` once we change its ssh auth to require one of the support yubikeys. Any suggestions on how to phrase that on top of these changes?

Reply:
Perhaps add a new integer field like then act on it, something like this (using `nopasswd`):

```
USERS=(
    (user=wicket group=wicket gecos='Wicket User' nopasswd=1)
    (user=support group=support gecos='Oxide Support'
        profiles=('Service Operator' 'Hotplug Management' 'Zone Management')
    )
)

for i in "${!USERS[@]}"; do
    nameref u=USERS[$i]
...

    if ((u.nopasswd)); then
            passwd -d "${u.user}"
    else
            passwd -N "${u.user}"
    fi
```